src/gasPriceOracle/oracle.ts

```diff
@@ -14,35 +14,36 @@ import * as polygon from "./adapters/polygon";
  * @param legacyFallback In the case of an unrecognised chain, fall back to type 0 gas estimation.
  * @returns Am object of type GasPriceEstimate.
  */
-export async function getGasPriceEstimate(
-  provider: providers.Provider,
-  chainId?: number,
-  legacyFallback = true
-): Promise<GasPriceEstimate> {
-  if (chainId === undefined) {
-    ({ chainId } = await provider.getNetwork());
-  }
+export async function getGasPriceEstimate(provider: providers.Provider, transport?: Transport): Promise<GasPriceEstimate> {
+  const { chainId } = await provider.getNetwork();
+
+  transport ??= http(); // @todo: Inherit URL from provider.
+  const chain = extractChain({ chains: Object.values(chains), id: chainId as any });
+  const viemProvider = createPublicClient({ chain, transport });
 
-  const gasPriceFeeds: { [chainId: number]: GasPriceFeed } = {
+  const gasPriceFeeds = {
     [CHAIN_IDs.ARBITRUM]: arbitrum.eip1559,
-    [CHAIN_IDs.BASE]: ethereum.eip1559,
-    [CHAIN_IDs.BOBA]: ethereum.legacy,
-    [CHAIN_IDs.LINEA]: linea.eip1559, // @todo: Support linea_estimateGas in adapter.
-    [CHAIN_IDs.MAINNET]: ethereum.eip1559,
-    [CHAIN_IDs.MODE]: ethereum.eip1559,
-    [CHAIN_IDs.OPTIMISM]: ethereum.eip1559,
     [CHAIN_IDs.POLYGON]: polygon.gasStation,
-    [CHAIN_IDs.ZK_SYNC]: ethereum.legacy,
-    [CHAIN_IDs.SCROLL]: ethereum.legacy,
-  };
+  } as const;
 
-  let gasPriceFeed = gasPriceFeeds[chainId];
-  if (gasPriceFeed === undefined) {
-    if (!legacyFallback) {
-      throw new Error(`No suitable gas price oracle for Chain ID ${chainId}`);
-    }
-    gasPriceFeed = chainIsOPStack(chainId) ? ethereum.eip1559 : ethereum.legacy;
+  let maxFeePerGas: bigint;
+  let maxPriorityFeePerGas: bigint;
+  if (gasPriceFeeds[chainId]) {
+		// declare function action<chain extends Chain | undefined>(client: Client<Transport, chain>): void
+		// action(viemProvider);
+		// https://github.com/wevm/viem/issues/2763
+    ({ maxFeePerGas, maxPriorityFeePerGas } = await gasPriceFeeds[chainId](viemProvider as any, chainId));
```

Comment:
@gsteenkamp89 @dohaki Example #2 where I can't get anything working without as typecast to `any`. The `Chain` input to `createPublicClient()` results in a very narrow inferred data type that I am struggling to know how to specify. This makes it really hard to pass that `PublicClient` instance around to other functions. The referenced github issue seems to acknowledge this and provides a solution. I don't understand what's going on there, though. Any clues?